command/fs_ls.go

```diff
@@ -68,6 +74,20 @@ func (f *FSListCommand) Run(args []string) int {
 		return 1
 	}
 
+	// If -job is specified, use random allocation, otherwise use provided allocation
+	allocID := ""
+	if job {
+		jobID := args[0]
+		allocs, _, err := client.Jobs().Allocations(jobID, nil)
+		if err != nil {
+			f.Ui.Error(fmt.Sprintf("Error querying API: %v", err))
+		}
+		r := rand.New(rand.NewSource(time.Now().UnixNano()))
+		allocID = allocs[r.Intn(len(allocs))].ID
```

Comment:
@ryanuber Or we could keep looking for an allocation whose client status is running. Just picking up an allocation from the last eval is not enough, the client might not have downloaded it, or it might not have started running, etc.